examples/scripts/stable_diffusion_tuning.py

```diff
@@ -0,0 +1,160 @@
+import os
+
+import numpy as np
+import requests
+import torch
+import torch.nn as nn
+from transformers import CLIPModel, CLIPProcessor
+
+from trl import DDPOConfig, DDPOTrainer, DefaultDDPOPipeline, DefaultDDPOScheduler
+
+
+class MLP(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.layers = nn.Sequential(
+            nn.Linear(768, 1024),
+            nn.Dropout(0.2),
+            nn.Linear(1024, 128),
+            nn.Dropout(0.2),
+            nn.Linear(128, 64),
+            nn.Dropout(0.1),
+            nn.Linear(64, 16),
+            nn.Linear(16, 1),
+        )
+
+    @torch.no_grad()
+    def forward(self, embed):
+        return self.layers(embed)
+
+
+class AestheticScorer(torch.nn.Module):
+    def __init__(self, dtype, cache=".", weights_fname="sac+logos+ava1-l14-linearMSE.pth"):
+        super().__init__()
+        self.clip = CLIPModel.from_pretrained("openai/clip-vit-large-patch14")
+        self.processor = CLIPProcessor.from_pretrained("openai/clip-vit-large-patch14")
+        self.mlp = MLP()
+        self.loadpath = os.path.join(cache, weights_fname)
+        if not os.path.exists(self.loadpath):
+            url = (
+                "https://github.com/christophschuhmann/"
+                f"improved-aesthetic-predictor/blob/main/{weights_fname}?raw=true"
+            )
+            r = requests.get(url)
+
+            with open(self.loadpath, "wb") as f:
+                f.write(r.content)
+
+        state_dict = torch.load(self.loadpath)
```

Comment:
I have a better suggestion for the name. You can store them on a HF model repo on the hub (i.e. upload the weights on a new fresh repo under your namespace) and use the `hf_hub_download` method from `huggingface_hub` library. Let us assume you have stored these weights under `myusername/ddpo-aestetic-scorer` repository, this block would look like:


```suggestion
    def __init__(self, dtype, hub_model_id="myusername/ddpo-aestetic-scorer"):
        super().__init__()
        self.clip = CLIPModel.from_pretrained("openai/clip-vit-large-patch14")
        self.processor = CLIPProcessor.from_pretrained("openai/clip-vit-large-patch14")
        self.mlp = MLP()
        cached_path = hf_hub_download(hub_model_id, "sac+logos+ava1-l14-linearMSE.pth")
        state_dict = torch.load(cached_path)
```

`hf_hub_download` method will directly save the downloaded file on the cache and return the absolute path of the cached file